### PR TITLE
Fix GetAccount

### PR DESCRIPTION
**Pull request by alexandru-calinoiu**
`GetAccount` was sending over a wrongfully formated url

### CODE REVIEW

File: erdgo/blockchain/proxy.go [hunk 1]
```diff
@@ -25,13 +25,23 @@ const (
 	getNetworkStatusEndpoint     = "network/status/%v"
 )
 
+type HTTPClient interface {
```
Comment (iulianpascalau) [inline erdgo/blockchain/proxy.go:28]: Missing comments on exported elements

File: erdgo/blockchain/proxy.go [hunk 2]
```diff
@@ -25,13 +25,23 @@ const (
...
+	Do(req *http.Request) (*http.Response, error)
+}
+
+var (
+	Client HTTPClient
```
Comment (iulianpascalau) [inline erdgo/blockchain/proxy.go:33]: Please do not use exported (static) instances. This is a bad practice as it is incredibly hard to test the instances that use exported variables.

File: erdgo/blockchain/proxy.go [hunk 3]
```diff
@@ -25,13 +25,23 @@ const (
...
+)
+
 // elrondProxy implements basic functions for interacting with an Elrond Proxy
 type elrondProxy struct {
 	proxyURL string
 }
 
 // NewElrondProxy initializes and returns an ElrondProxy object
 func NewElrondProxy(url string) *elrondProxy {
+	Client = http.DefaultClient
```
Comment (iulianpascalau) [inline erdgo/blockchain/proxy.go:43]: please declare this as an unexported field in elrondProxy.

File: erdgo/blockchain/proxy.go [hunk 4]
```diff
@@ -205,8 +215,7 @@ func (ep *elrondProxy) getHTTP(endpoint string) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	client := http.DefaultClient
-	response, err := client.Do(request)
+	response, err := Client.Do(request)
```
Comment (iulianpascalau) [inline erdgo/blockchain/proxy.go:218]: why this change?

File: erdgo/blockchain/proxy.go [hunk 5]
```diff
@@ -66,7 +76,7 @@ func (ep *elrondProxy) GetAccount(address addressHandler) (*data.Account, error)
 	if !address.IsValid() {
 		return nil, ErrInvalidAddress
 	}
-	endpoint := fmt.Sprintf(accountEndpoint, address)
+	endpoint := fmt.Sprintf(accountEndpoint, address.AddressAsBech32String())
```
Comment (iulianpascalau) [inline erdgo/blockchain/proxy.go:79]: :+1:

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: erdgo/blockchain/proxy.go (hunk 2 above)
Comment (alexandru-calinoiu) [reply erdgo/blockchain/proxy.go:33]: I've actually did in order to be able to write a test, the 2 methods `get` and `post` where using client.Default that made it impossible to test.

As an alternative I could inject in the constructor and don't export it, does this sound better?
(will actually have to make a new constructor so I don't break existing clients)

File: erdgo/blockchain/proxy.go (hunk 4 above)
Comment (alexandru-calinoiu) [reply erdgo/blockchain/proxy.go:218]: Injecting a mock http client allows this method to be tested.

File: erdgo/blockchain/proxy.go [hunk 6]
```diff
@@ -25,15 +25,25 @@ const (
...
+
 // elrondProxy implements basic functions for interacting with an Elrond Proxy
 type elrondProxy struct {
 	proxyURL string
+	client   HTTPClient
 }
 
 // NewElrondProxy initializes and returns an ElrondProxy object
-func NewElrondProxy(url string) *elrondProxy {
+func NewElrondProxy(url string, client HTTPClient) *elrondProxy {
```
Comment (iulianpascalau) [inline erdgo/blockchain/proxy.go:39]: :+1:

File: erdgo/blockchain/proxy.go [hunk 7]
```diff
@@ -25,15 +25,25 @@ const (
...
+	if client == nil {
```
Comment (iulianpascalau) [inline erdgo/blockchain/proxy.go:40]: instead of this check we use the check.IfNill(client) call provided that he HTTPClient implements IsInterfaceNil.

**Review by iulianpascalau — COMMENTED** [review]
Looks better :+1:

**Review by iulianpascalau — APPROVED** [review]

File: erdgo/tests/proxy_test.go [hunk 8]
```diff
@@ -0,0 +1,43 @@
+package tests
+
+import (
+	"bytes"
+	"github.com/ElrondNetwork/elrond-sdk/erdgo/blockchain"
```
Comment (bogdan-rosianu) [inline erdgo/tests/proxy_test.go:5]: can you run goimports?

File: erdgo/blockchain/proxy.go [hunk 9]
```diff
@@ -223,7 +233,12 @@ func (ep *elrondProxy) getHTTP(endpoint string) ([]byte, error) {
 
 func (ep *elrondProxy) postHTTP(endpoint string, data []byte) ([]byte, error) {
 	url := fmt.Sprintf("%s/%s", ep.proxyURL, endpoint)
-	response, err := http.Post(url, "", strings.NewReader(string(data)))
+	request, err := http.NewRequest("POST", url, strings.NewReader(string(data)))
```
Comment (bogdan-rosianu) [inline erdgo/blockchain/proxy.go:236]: you could have used `http.MethodPost` instead of `"POST"`

Comment (bogdan-rosianu) [reply erdgo/blockchain/proxy.go:236]: also you could have used `bytes.NewReader(data)` instead of casting to string and creating a string reader

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]
